arduino-core/src/processing/app/Sketch.java

```diff
@@ -355,8 +355,8 @@ public void saveAs(File newFolder) throws IOException {
     // Copy the data folder (this may take a while.. add progress bar?)
     if (getDataFolder().exists()) {
       File newDataFolder = new File(newFolder, "data");
-      // Create the data folder
-      if (!newDataFolder.mkdirs()) {
+      // Check if data folder exits, if not try to create the data folder
+      if (!(newDataFolder.exists() || newDataFolder.mkdirs())) {
```

Comment:
I would think that `if (!newDataFolder.exists() && !newDataFolder.mkdirs())` is a bit easier to read, would you agree?